src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -935,7 +959,9 @@ func (s *peersSource) processReaders(
 	newFulfilled.AddRanges(indexBlock.Fulfilled())
 
 	// Replace index block for default index volume type.
+	resultLock.Lock()
```

Comment:
is it possible to embed the locking in the impl of the results?

Reply:
also do you need read locking?

Reply:
Yea we could maybe make the `runResult` struct in the fs bootstrapper shared across all bootstrappers. 

Might be a good idea since then we'll have a consistent way of locking results wherever we add concurrency in the future as well.